core/lib/resources/presence/sentry.js

```diff
@@ -1,159 +1,270 @@
+// we compare existing sentry names to reply names
+// and clear whatever we have that no longer exists. 
+Sentry.prototype._purgeGoneSentries = function(replies, repliesKeys) {
+  var self = this,
+      sentriesGone = _.difference(this.sentryNames(), repliesKeys);
 
-  var message = {
-    name: name,
-    alive: true,
-    host: host,
-    port: port,
-    expiration: expiration
-  };
+  sentriesGone.forEach(function(name) {
+    self._purgeSentry(name);
+  });
+};
 
-  this.sentries[name] = message;
+// Listening for new pub sub messages from redis. 
+// As of now, we only care about new sentries going online. 
+// Everything else gets inferred based on time. 
+Sentry.prototype._startListening = function() {
+  var self = this;
 
-  // Specific check
-  if (options.save === false) return;
+  if (!this._listener) {
+    this._listener = function(channel, message) {
+      if (channel !== redisSentriesKey) {
+        return;
+      }
+      self._saveMessage(parseJSON(message));
+    };
 
-  Persistence.persistHash(Sentry.channel, this.name, message);
-  Persistence.publish(Sentry.channel, message);
-};
+    Persistence.pubsub().on('message', this._listener);
+  }
 
-Sentry.prototype.start = function(callback) {
-  var sentries = this.sentries;
-  var self = this;
-  if (this.timer) {
-    return;
+  if (!this._subscribed) {
+    this._subscribed = true;
+    Persistence.pubsub().subscribe(redisSentriesKey);
   }
-  logging.info('#presence - #sentry - starting', this.name);
-  this._listen();
-  this.publishKeepAlive();
-  this.loadAll(callback);
-  this.timer = setTimeout(this._run.bind(this), Math.floor(Sentry.expiry/2));
 };
 
-Sentry.prototype.stop = function() {
-  logging.info('#presence - #sentry stopping', this.name);
-  if (this.timer) {
-    clearTimeout(this.timer);
-    delete this.timer;
+Sentry.prototype._stopListening = function() {
+  if (this._subscribed) {
+    Persistence.pubsub().unsubscribe(redisSentriesKey);
+    delete this._subscribed;
   }
-  Persistence.pubsub().unsubscribe(Sentry.channel);
-  delete this.subscribed;
-  if (this.listener) {
-    Persistence.pubsub().removeListener('message', this.listener);
-    delete this.listener;
+
+  if (this._listener) {
+    Persistence.pubsub().removeListener('message', this._listener);
+    delete this._listener;
   }
-  this.sentries = {};
 };
 
-Sentry.prototype.isValid = function(name) {
-  var message = this.sentries[name];
-  var valid = (message && message.expiration && (message.expiration >= Date.now()));
-  var expiration = (message && message.expiration && (message.expiration - Date.now()));
+Sentry.prototype._saveMessage = function(message) {
+  if (message && message.name) {
+    logging.debug('#presence - sentry.save', message.name, messageExpiration(message));
+    this.sentries[message.name] = message;
+  }
+};
+
+Sentry.prototype._refresh = function() {
+  var interval = Math.floor(this._refreshInterval/2);
```

Comment:
Again, the refresh interval is small enough. (at least 5 times within the expiry period) So no need for /2